lib/framework/framework.dart

```diff
@@ -0,0 +1,289 @@
+  final Map<Screen, List<Element>> _contents = {};
+
+  void load(Screen screen) {
+    if (current != null) {
+      current.exiting();
+      pageStatus.removeAll();
+      _contents[current] = mainElement.element.children.toList();
+      mainElement.element.children.clear();
+    } else {
+      mainElement.element.children.clear();
+    }
+
+    current = screen;
+
+    if (_contents.containsKey(current)) {
+      mainElement.element.children.addAll(_contents[current]);
+    } else {
+      current.createContent(this, mainElement);
+    }
+
+    current.entering();
+    pageStatus.addAll(current.statusItems);
+
+    updatePage();
+  }
+
+  void updatePage() {
+    // nav
+    for (Element element in querySelectorAll('#main-nav a')) {
+      CoreElement e = new CoreElement.from(element);
+      bool isCurrent = current.ref == element.attributes['href'];
+      e.toggleClass('active', isCurrent);
+    }
+
+    // status
+    CoreElement helpLink = new CoreElement.from(querySelector('#docsLink'));
+    HelpInfo helpInfo = current.helpInfo;
+    if (helpInfo == null) {
+      helpLink.hidden(true);
+    } else {
+      helpLink.clear();
+      helpLink.add([
+        span(text: '${helpInfo.title} '),
+        span(c: 'octicon octicon-link-external small-octicon'),
+      ]);
+      helpLink.setAttribute('href', helpInfo.url);
+      helpLink.hidden(false);
+    }
+  }
+
+  void showError(String title, [dynamic error]) {
+    PFlash flash = new PFlash();
+    flash.addClose().click(clearError);
+    flash.add(span(text: title));
+    if (error != null) {
+      flash.add(new CoreElement('br'));
+      flash.add(span(text: '$error'));
+    }
+
+    CoreElement errorContainer =
+        new CoreElement.from(querySelector('#error-container'));
+    errorContainer.add(flash);
+  }
+
+  void clearError() {
+    querySelector('#error-container').children.clear();
+  }
+}
+
+class StatusLine {
+  final CoreElement element;
+  final List<StatusItem> _items = [];
+
+  StatusLine(this.element);
+
+  void add(StatusItem item) {
+    _items.add(item);
+
+    _rebuild();
+  }
+
+  void _rebuild() {
+    element.clear();
+
+    if (_items.isNotEmpty) {
+      element.add(_items.first.element);
+
+      for (StatusItem item in _items.sublist(1)) {
+        element.add(new SpanElement()
+          ..text = '•'
+          ..classes.add('separator'));
+        element.add(item.element);
+      }
+    }
+  }
+
+  void remove(StatusItem item) {
+    _items.remove(item);
+
+    _rebuild();
+  }
+
+  void addAll(List<StatusItem> items) {
+    _items.addAll(items);
+
+    _rebuild();
+  }
+
+  void removeAll() {
+    _items.clear();
+    _rebuild();
+  }
+}
+
+void toast(String message) {
+  // TODO:
+  print(message);
+}
+
+abstract class Screen {
+  final String name;
+  final String id;
+  final String iconClass;
+
+  final Property<bool> _visible = new Property(true);
+
+  final List<StatusItem> statusItems = [];
+
+  Screen(this.name, this.id, [this.iconClass]);
```

Comment:
nit: probably would be nice to tweak this to use all named arguments as everything is a string argument so could be easy to otherwise switch arg order.